docs/HowToFAQ/FAQFailingPost.md

```diff
@@ -0,0 +1,30 @@
+---
+title: I can assert a condition right before a return, so why does the postcondition fail?
+---
+
+## Question
+
+I can assert a condition right before a return, so why does the postcondition fail?
+
+## Answer
+
+There can be lots of reasons why a postcondition might fail to verify, but if you can prove the same predicate just before a return, 
+a typical reason is that there are other exit paths from the method or function. There might be other `return` statements, but a
+harder to spot exit path is the `:-` let-or-fail assignment.
+Here is a sketch of that kind of problem:
+
+```
+method test(MyClass o) returns (r: Result<int>)
+  modifies o;
+  ensures o.ok == true;
```

Comment:
Could we make this a self-contained example?

Reply:
(Possibly by importing the standard library)

Reply:
Added a reference to the Result type in the library, but I'd rather not include the whole datatype because the code is rather lengthy.

Reply:
Added that, with additional text